### ISSUE TITLE
gitinfo2 conflicts with datetime package

### ISSUE THREAD

**Issue by maxheld83** [Hightor/gitinfo2#9]
I called `\usepackage{datetime}` _before_ \usepackage{gitinfo2}, as advised.

I get this:

```
! Undefined control sequence.
<argument> ...Date }{-}{\gitInf@year }{\gitInf@md 
                                                  } \StrCut {\gitInf@md }{-}...
l.198 }{}
         %
The control sequence at the end of the top line
of your error message was never \def'ed. If you have
misspelled it (e.g., `\hobx'), type `I' and the correct
spelling (e.g., `I\hbox'). Otherwise just continue,
and I'll forget about whatever was undefined.
```

The offending line 198 appears to be in `/usr/local/texlive/2014/texmf-dist/tex/latex/gitinfo2/gitinfo2.sty`:

```
}{}%
```

Not sure what's going on, or whether this might be on the datetime-end, just thought it'd bring it up.

**Comment by longborough** [comment]
That's going to take some finding. I'm sorry, but as I'm away from home at the moment, it's going to have to wait till I get back.

**Comment by maxheld83** [comment]
thanks a bunch – I greatly appreciate your generous work on this package; I just thought I'd list all of the things I ran into.

**Comment by longborough** [comment]
I had a quick try on my laptop here in the hotel, and couldn't reproduce this. I'm using the feature in a number of 'live' documents I'm currently working on, and it just doesn't break.

If you have time and inclination, could you work it down to an absolute minimum document that fails, and post it here, along with your gitHeadInfo.gin?

**Comment by maxheld83** [comment]
Sure!
See below – didn't even have to actually _use_ gitinfo2, just the preambs seem to cause the errors. Compiles just fine if line 2 is commented out.

```
\documentclass{article}
\usepackage{datetime} % compiles fine if this is commented out
\usepackage{xcolor}
\usepackage{gitinfo2}

\begin{document}

Test.

\end{document}
```

Log:

```
E: /usr/local/texlive/2014/texmf-dist/tex/latex/gitinfo2/gitinfo2.sty:198 Undefined control sequence }{}
E: /usr/local/texlive/2014/texmf-dist/tex/latex/gitinfo2/gitinfo2.sty:198 Missing \begin{document}. }{}
E: /usr/local/texlive/2014/texmf-dist/tex/latex/gitinfo2/gitinfo2.sty:198 Undefined control sequence }{}
E: /usr/local/texlive/2014/texmf-dist/tex/latex/gitinfo2/gitinfo2.sty:198 Undefined control sequence }{}
E: /usr/local/texlive/2014/texmf-dist/tex/latex/gitinfo2/gitinfo2.sty:198 Undefined control sequence }{}
E: /usr/local/texlive/2014/texmf-dist/tex/latex/gitinfo2/gitinfo2.sty:198 Undefined control sequence }{}
E: /usr/local/texlive/2014/texmf-dist/tex/latex/gitinfo2/gitinfo2.sty:198 Undefined control sequence }{}
E: /usr/local/texlive/2014/texmf-dist/tex/latex/gitinfo2/gitinfo2.sty:198 Undefined control sequence }{}
E: /usr/local/texlive/2014/texmf-dist/tex/latex/gitinfo2/gitinfo2.sty:198 Undefined control sequence }{}

W: /usr/local/texlive/2014/texmf-dist/tex/latex/gitinfo2/gitinfo2.sty:0 I can't find the file '.git/gitHeadInfo.gin'.

9 errors, 1 warning, 0 badboxes
```

gitHeadInfo.gin looks like this:

```
\usepackage[%
        shash={9ba9053},
        lhash={9ba9053da0ceb05b0bad19723b0e2c0a7e3e9d3a},
        authname={Maximilian Held},
        authemail={maximilian.held83@gmail.com},
        authsdate={2014-11-08},
        authidate={2014-11-08 22:35:15 +0100},
        authudate={1415482515},
        commname={Maximilian Held},
        commemail={maximilian.held83@gmail.com},
        commsdate={2014-11-08},
        commidate={2014-11-08 22:35:15 +0100},
        commudate={1415482515},
        refnames={ (HEAD, master)}
    ]{gitsetinfo}
```

Also, oddly, `gitHeadInfo.gin` always gets placed in the project root, not inside `.git`, as per #10

**Comment by longborough** [comment]
OK, now I think I know what's going on. This is a combination of (a) not being able to find .git/gitHeadInfo.gin; and (b) a bug when that happens (dependent packages called in gitexinfo rather than gitinfo2.

As soon as I can, I'll send a fix to github, and shortly after to CTAN.

Meanwhile, we need to fix #10

**Comment by longborough** [comment]
Fixed by commit a1cc425 for release 2.0.5
